src/state/direct/future.rs

```diff
@@ -1,10 +1,16 @@
+use std::num::NonZeroU32;
+
 use super::RateLimiter;
 use crate::{
     clock,
     state::{DirectStateStore, NotKeyed},
-    Jitter,
+    Jitter, NegativeMultiDecision,
 };
-use futures_timer::Delay;
+use {futures_timer::Delay, thiserror::Error};
+
+#[derive(Debug, Error, Clone)]
+#[error("required number of cell {} exceeds bucket's capacity", _0)]
```

Comment:
Hm. I'd really love to avoid pulling in the `thiserror` crate here - impl'ing Display for the type should be pretty easy, and it avoids a proc macro dependency... I've been hunting those down over the past half hour and am pretty close to having the non-dev build be free of deps on `syn` (:

Reply:
Sure thing. Whats the reasoning to avoid the `syn` crate? Is it to reduce compile time?

Reply:
That's part of it; I've also seen weird multiple dependency situations where several crates ended up depending on different versions of `sym` and such. I'm a fan of `thiserror`, but for small error type scenarios (only one with a straightforward Display impl), I feel it's easier to do without (: